### PR TITLE
Store evaluations from asset daemon

### PR DESCRIPTION
**Pull request by johannkm**
Keep an evaluation id in the cursor, and write asset evaluations on each iteration

Added a backcompat test for cursors without an evaluation id

Added a check that the new table exists so that people can upgrade and the daemon won't crash

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14295" target="_blank">Graphite</a> 
    * **PR #14319** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14319" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/14319?utm_source=stack-comment).

File: python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/test_asset_daemon.py [hunk 1]
```diff
@@ -41,9 +42,16 @@ def sort_run_request_key_fn(run_request):
 
 @pytest.fixture
 def daemon_not_paused_instance():
-    instance = DagsterInstance.ephemeral()
-    set_auto_materialize_paused(instance, False)
-    yield instance
+    with instance_for_test(
+        overrides={
+            "run_launcher": {
```
[the remark this answers is not in the document]
Comment (johannkm) [reply python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/test_asset_daemon.py:47]: previously this was the ephemeral instance which doesn't have schedule storage. Using the default run launcher doesn't work with the in process code locations (which are much faster than subprocess grpc severs for these tests)

File: python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py [hunk 2]
```diff
@@ -341,15 +344,22 @@ def empty(cls) -> "AssetReconciliationCursor":
...
+        check.invariant(len(data) in [3, 4], "Invalid serialized cursor")
+
         (
             latest_storage_id,
             serialized_materialized_or_requested_root_asset_keys,
             serialized_materialized_or_requested_root_partitions_by_asset_key,
-        ) = json.loads(cursor)
+        ) = data[:3]
+
+        evaluation_id = data[3] if len(data) == 4 else None
```
[the remark this answers is not in the document]
Comment (johannkm) [reply python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py:361]: I was thinking of adding logging when this happens but we can probably do that either way

[the remark this answers is not in the document]
Comment (johannkm) [reply python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py:361]: easier to do the logging from the daemon though so I might leave it for now

File: python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/test_asset_daemon.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (OwenKephart) [reply python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/test_asset_daemon.py:47]: gotcha makes sense

**Pull request merged**